packages/vm/src/context.rs

```diff
@@ -116,13 +148,27 @@ pub(crate) fn move_into_context<S: Storage, Q: Querier>(target: &mut Ctx, storag
     b.querier = Some(querier);
 }
 
+pub fn get_gas_status<S: Storage, Q: Querier>(ctx: &mut Ctx) -> &mut GasState {
+    &mut get_context_data::<S, Q>(ctx).gas_state
+}
+
+pub fn try_consume_gas<S: Storage, Q: Querier>(ctx: &mut Ctx, used_gas: u64) -> VmResult<()> {
+    let gas_status = get_gas_status::<S, Q>(ctx);
+    gas_status.use_gas(used_gas);
+    if gas_status.out_of_gas() {
+        Err(VmError::GasDepletion)
+    } else {
+        Ok(())
+    }
+}
```

Comment:
Doesn't this implementation bring `ContextData.gas_state` and the Wasmer gas usage out of sync?

Reply:
It does. I have a fix locally but i didn't push it because it depends on #245

Reply:
since #245 is now merged to 0.9, i merged 0.9 to #359, and merged that back into this PR. I then added a commit with the fix to this issue, so you can review the latest commit to resolve this conversation :)

Reply:
Arg, can we just get everything into one PR and `git rebase 0.9` that one?

Reply:
This PR now strictly supersedes #359 so we can just make this point to `0.9`. But we're really done here.
I don't see a need for any rebasing to be honest.

Reply:
okay, thanks

Reply:
I mean, merging this to 0.9 is (at the moment) completely equivalent to merging this into #359 and then merging #359 into 0.9

Reply:
The gas consumption in wasmer and the external service (Cosmos SDK) are fully synced, and wasmer is informed about gas consumed externally.